services/indexer/notifier.go

```diff
@@ -30,6 +31,10 @@ func NewNotifier() base.Notifier {
 	return &indexerNotifier{}
 }
 
+func init() {
+	notification.RegisterNotifier(NewNotifier())
+}
```

Comment:
How do we make sure those `init` functions have been executed?

I mean, I can't find `import code.gitea.io/gitea/services/indexer`.

Reply:
Isn't that actually pretty simply?
Just declare one file that is **always** loaded and at the bottom of the call hierarchy that loads all packages with an `init` function?
Something like `main.go`?

Reply:
Yeah, it's easy to fix it, but there are some problems to consider:

1. Some service packages have `Init()`, should we register the notifier in it?
2. If yes, both `init()` and `Init()` are used, I think it's easy to confuse and make mistakes.
3. If no, so all packages that have a notifier should define a `init()` and be imported by `import _ "code.gitea.io/gitea/services/indexer"`, even if it has `Init()` and already has been imported.
4. One more case, Actions has to use `Init()` to register the notifier, because it needs to read the setting and decide if need to register. See https://github.com/go-gitea/gitea/pull/21937/files#diff-d3da948e9835f082186a9f16e8f27edc79b3113e6deb092ea308eb4fa8d96015:
```go
func Init() {
	if !setting.Actions.Enabled {
		return
	}

	jobEmitterQueue = queue.CreateUniqueQueue("actions_ready_job", jobEmitterQueueHandle, new(jobUpdate))
	go graceful.GetManager().RunWithShutdownFns(jobEmitterQueue.Run)

	notification.RegisterNotifier(NewNotifier())
}
```

As you can see, I followed this refactor in the actions branch, so I'm not against it.

Reply:
Done.